src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2325,9 +2321,13 @@ bool IsVirtualNeededByTypeDueToPreservedScope (MethodDefinition method)
 			return false;
 		}
 
-		bool IsMethodNeededByTypeDueToPreservedScope (MethodDefinition method)
+		/// <summary>
+		/// Returns true if any of the base methods of <paramref name="method" /> is defined in an assembly that is not trimmed (i.e. action!=trim)
+		/// </summary>
+		/// <remarks>This is very similar to <see cref="IsVirtualNeededByTypeDueToPreservedScope(MethodDefinition)"/>, except this also checks if the base method is an interface.</remarks>
+		bool IsVirtualNeededByInstantiatedTypeDueToPreservedScope (MethodDefinition method)
 		{
-			if (!method.IsVirtual)
+			if (!(method.IsVirtual || method.IsStatic))
```

Comment:
Maybe add a comment that this handles static interface methods (it's not obvious why statics are interesting here otherwise).